docs/design_proposals/kaniko-proxy-setting.md

```diff
@@ -0,0 +1,122 @@
+# Title
+
+* Author(s): Prashant Arya
+* Design Shepherd: Tejal Desai
+* Date: 3rd April 2018
+* Status: 
+
+## Background
+
+At present if you run skaffold in seperate lab environment which doesn't have direct internet access
+and need proxy setting then skaffold can directly access the envrionment variable of the lab and can
+access the artifactory or registry but on the other hand builder like kaniko which spins up a seperate 
+pod fails to build and push the built image to registry as it doesn't have proxy environment set. 
+Hence skaffold run fails.
+
+To overcome this problem we can set proxy in environment variable of kaniko pod defination and get the
+skaffold running. We can take proxy variable in kaniko config section.
+
+ 
+Please provide a brief explanation for the following questions:
+
+1. Why is this required? So that skaffold can run in lab environment where access to internet is via proxy
+2. If this is a redesign, what are the drawbacks of the current implementation? I don't have idea about the other builder,
+3. Is there any another workaround, and if so, what are its drawbacks? I can't think of drawback
+4. Mention related issues, if there are any. NA
+
+Here is an example snippet for a new feature:
+
+___
+Setting proxy variable in pod definition 
+```yaml
+Containers: []v1.Container{
+                {
+                    Name:            constants.DefaultKanikoContainerName,
+                    Image:           image,
+                    Args:            args,
+                    ImagePullPolicy: v1.PullIfNotPresent,
+                    Env: []v1.EnvVar{
+                        {
+                        Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+                        Value: "/secret/kaniko-secret",
+                        }
+                        {
+                        Name:  "http_proxy",
+                        Value: "somevalue",
+                        }
+                        {
+                        Name:  "https_proxy",
+                        Value: "somevalue",
+                        }
+                    },
+                },
+```
+Setting the proxy would give kaniko pod internet access. Where it can contact gcr or linux update server(any install command).
+___
+
+## Design
+
+Please describe your solution. Please list any:
+
+* new config changes
+* interface changes
+* design assumptions
+
+For a new config change, please mention:
+
+* Is it backwards compatible? If not, what is the deprecation policy? No idea 
+  
+```yaml
+// ClusterDetails *beta* describes how to do an on-cluster build.
+type ClusterDetails struct {
+    
+    // http_proxy     
+    http_proxy string `yaml:"http_proxy,omitempty"`
+
+    // https_proxy
+    https_proxy string `yaml:"http_proxys,omitempty"`
```

Comment:
hi @prary, are https_proxy and http_proxys the new config values?

Reply:
Adding these 2 fields is not a big issue. I think the team is ok with it!
Would you be able to submit a PR?

Reply:
Yes I make the changes and raise a PR for the same

Reply:
Is there any deadline to raise PR?